### DISCUSSION TITLE
Broken language analysis in GDScript since introduction of "Scene unique nodes"

### DISCUSSION THREAD

**Discussion by rsubtil** [github-linguist/linguist#6108] [Classification] [answered]
Godot introduced a new feature called "[Scene unique nodes](https://docs.godotengine.org/en/stable/tutorials/scripting/scene_unique_nodes.html)" where nodes can be accessed differently with the `%` operator.

This breaks the syntax highlighting.

Before:
```gdscript
var my_node = $/Node1/Node2/MyNode

func _ready() -> void:
    my_node.visible = false
```

After:
```gdscript
var my_node = $"%MyNode"

func _ready() -> void:
    my_node.visible = false
```

**Comment by lildude** [comment, marked as answer]
Linguist has no control over the way upstream grammars behave. Please open an issue with the upstream grammar maintainers. Any updates to grammars will be pulled in when the next Linguist release is made.

**Reply by rsubtil** [reply]
Thanks!
